Only enable DropDownList mouse scrolling when dropped or shift keyed pressed.

LGM-Doyle opened the pull request:
Addresses the first part of issue #1573.

LGM-Doyle committed:
Add SetOnlyMouseScrollWhenDrop to drop down lists.

It allows preventing mouse scrolling when the drop down list is not
open.

LGM-Doyle committed:
Use SetOnlyMouseScrollWhenDropped() in ProductionWnd and SidePanel.

Used for the QuantitySelectors and focus drop downs.

LGM-Doyle committed:
Change comment for review #1577.

Change only -> only when.

geoffthemedio reviewed (commented):
Should probably also do options wnd droplists until they it doesnt have scrollbars.

LGM-Doyle committed:
Use SetOnlyMouseScrollWhenDropped() in OptionsWnd.

As per review #1577

LGM-Doyle committed:
In DropDownList ModalListPicker capture mousewheel events from child ListBox.

Redirection of mouse events and keyboard events for modal windows are
handled by code in GUI.cpp that directs the keyboard events to the top
level modal window and the mouse events to the child list box.

This commit adds an event filter to capture the mouse wheel events from
the child list box.

LGM-Doyle committed:
Remove include Logger from GG code.

Found in review #1577.

LGM-Doyle committed:
Make DropDownList mouse wheel scroll the dropped listbox.

Bug found in review #1577.

LGM-Doyle committed:
Make DropDownList accept mouse wheel events in anchor window.

Bug found in review #1577.
